src/extension/src/generationExperience.ts

```diff
@@ -146,7 +147,37 @@ export class GenerationExperience extends WizardServant {
 
     // Resource groups should be created before other deploy methods execute
     Promise.all(resourceGroupQueue).then(() => {
-      // Create ASP (this PR) and create web app (future PR) will be called here
+      if (payload.selectedAppService) {
+        serviceQueue.push(
+          GenerationExperience.Telemetry.callWithTelemetryAndCatchHandleErrors(
+            TelemetryEventName.AppServiceDeploy,
+            // tslint:disable-next-line: no-function-expression
+            async function(this: IActionContext): Promise<void> {
```

Comment:
Should we declare this function somewhere outside and then pass the function name into `callWithTelemetryAndCatchHandleErrors` for cleaner code?

Reply:
will handle this in some other PR. I want things to be consistent with the way that we deploy functions and cosmos for now

Reply:
Okay!